patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -791,13 +791,22 @@
 +     * @param armor the armor itemstack
 +     */
 +    public void onHorseArmorTick(World world, net.minecraft.entity.EntityLiving horse, ItemStack armor) {}
++    
++    /**
++     * @return This Item's TEISR, this method should NOT return a new instance.  Must not be null.
```

Comment:
I think this might be a bit unclear, someone could take "should not return a new instance" to mean "don't do anything custom" which would make it quite a useless method indeed.

Maybe something like "the return value of this method should be constant for the lifetime of the client" or similar